docs/Gamification_extension_mechanism.rst

```diff
@@ -0,0 +1,175 @@
+
+Domain configuration
+~~~~~~~~~~~~~~~~~~~~
+-  ``zone-name``: The name of the new domain.
+-  ``zone-description``: The description of the new domain.
+-  ``zone-priority``: The domain priority.
+
+
+.. code:: xml
+
+    <external-component-plugins>
+        <target-component>org.exoplatform.addons.gamification.service.setting.zone.ZoneRegistry</target-component>
+        <component-plugin>
+            <name>developer.zone</name>
+            <set-method>addPlugin</set-method>
+            <type>org.exoplatform.addons.gamification.service.setting.zone.model.ZoneConfig</type>
+            <init-params>
+                <value-param>
+                    <name>zone-name</name>
+                    <value>Development</value>
+                </value-param>
+                <value-param>
+                    <name>zone-description</name>
+                    <value>Development</value>
+                </value-param>
+                <value-param>
+                    <name>zone-priority</name>
+                    <value>5</value>
+                </value-param>
+            </init-params>
+        </component-plugin>
+    </external-component-plugins>
+
+===============
+Add new badges
+===============
+
+To add new badge to the gamification configuration a simple external component plugin should be added to your addon configuration, this plugin should define the following information:
+
+Badge configuration
+~~~~~~~~~~~~~~~~~~~
+-  ``badge-title``: The name of the new badge.
+-  ``badge-description``: The description of the new badge.
+-  ``badge-domain``: The domain of the new badge.
+-  ``badge-icon``: The icon of the badge.
+-  ``badge-neededScore``: The needed score that the user should have to get the badge.
+-  ``badge-enable``: Define if the rule is enabled or not
+
+
+
+.. code:: xml
+
+    <external-component-plugins>
+        <target-component>org.exoplatform.addons.gamification.service.setting.badge.BadgeRegistry</target-component>
+        <component-plugin>
+            <name>badge.SocialNovice</name>
+            <set-method>addPlugin</set-method>
+            <type>org.exoplatform.addons.gamification.service.setting.badge.model.BadgeConfig</type>
+            <init-params>
+                <value-param>
+                    <name>badge-title</name>
+                    <value>Social Novice</value>
+                </value-param>
+                <value-param>
+                    <name>badge-description</name>
+                    <value>Growing your network and ramping up your interactions</value>
+                </value-param>
+                <value-param>
+                    <name>badge-domain</name>
+                    <value>Social</value>
+                </value-param>
+                <value-param>
+                    <name>badge-icon</name>
+                    <value>socialNovice.png</value>
```

Comment:
Should clarify where/how this image should be packaged by 3rd party developers that want to customize badges